contracts/HydraDelegation/DelegationPoolLib.sol

```diff
@@ -159,19 +209,152 @@ library DelegationPoolLib {
+     * @param pool the DelegationPool to clean the historical data from
+     * @param account the address of the  delegator to clean the historical data from
+     */
+    function cleanDelegatorHistoricalData(DelegationPool storage pool, address account) internal {
+        delete pool.delegatorsParamsHistory[account];
+    }
+
+    /**
+     * @notice Gets the delegator's reward at past moment based on a given epoch number and balance change index
+     * @param pool the DelegationPool to query the claimable rewards from
+     * @param delegator Address of the delegator
+     * @param epochNumber Epoch number
+     * @param balanceChangeIndex Index of the balance change
+     */
+    function _getRewardParams(
+        DelegationPool storage pool,
+        address delegator,
+        uint256 epochNumber,
+        uint256 balanceChangeIndex
+    ) private view returns (uint256 rps, uint256 balance, int256 correction) {
+        uint256 rewardPerShare = pool.historyRPS[epochNumber].value;
+        (uint256 balanceData, int256 correctionData) = _getAccountParams(
+            pool,
+            delegator,
+            epochNumber,
+            balanceChangeIndex
+        );
+
+        return (rewardPerShare, balanceData, correctionData);
+    }
+
+    /**
+     * @notice Saves the RPS for the given staker for the epoch
+     * @param pool the DelegationPool to save the RPS for
+     * @param rewardPerShare Amount of tokens to be withdrawn
+     * @param epochNumber Epoch number
+     */
+    function _saveEpochRPS(DelegationPool storage pool, uint256 rewardPerShare, uint256 epochNumber) private {
+        require(rewardPerShare > 0, "rewardPerShare must be greater than 0");
+
+        RPS memory poolRPS = pool.historyRPS[epochNumber];
+        require(poolRPS.value == 0, "RPS already saved");
+
+        pool.historyRPS[epochNumber] = RPS({value: uint192(rewardPerShare), timestamp: uint64(block.timestamp)});
+    }
+
+    /**
+     * @notice Gets the account specific pool params for the given epoch
+     * @param pool the DelegationPool to query the account params from
+     * @param delegator Address of the vest manager
+     * @param epochNumber Epoch number
+     * @param paramsIndex Index of the params
+     */
+    function _getAccountParams(
+        DelegationPool storage pool,
+        address delegator,
+        uint256 epochNumber,
+        uint256 paramsIndex
+    ) private view returns (uint256 balance, int256 correction) {
+        if (paramsIndex >= pool.delegatorsParamsHistory[delegator].length) {
+            revert DelegateRequirement({src: "vesting", msg: "INVALID_PARAMS_INDEX"});
+        }
+
+        DelegationPoolDelegatorParams memory params = pool.delegatorsParamsHistory[delegator][paramsIndex];
+        if (params.epochNum > epochNumber) {
+            revert DelegateRequirement({src: "vesting", msg: "LATE_BALANCE_CHANGE"});
+        } else if (params.epochNum == epochNumber) {
```

Comment:
Can we optimize this to have less if statements? else if (params.epochNum != epochNumber && paramsIndex != pool.delegatorsParamsHistory[delegator].length - 1) { do the logic needed}

Reply:
Good catch